### ISSUE TITLE
Ensure that FMP registration passes ssn metadata correctly

### ISSUE THREAD

**Issue by baltes** [department-of-veterans-affairs/va.gov-team#95292]
Confirm with PEGA team whether this is just a metadata change.

As a FMP PA, I need PEGA to pull SSN rather than VA File Number onto my screen when both are included.

**Include feature flag for testing on staging**

- If SSN is populated and file number is NOT populated, send SSN in the metadata
- If SSN is populated AND file number is populated, send SSN in the metadata
- If SSN is NOT populated and file number is populated, send file number in the metadata

**Comment by michaelclement** [comment]
Current backend logic for the FMP1 metadata:
![Image](https://github.com/user-attachments/assets/38be9751-f8b5-42fd-85fd-8649e4b48396)

**Comment by reiting** [comment]
This PR appears to be stuck. It was approved but for some reason 3 checks aren't running. I've put in a ticket with platform support and will hopefully get it moving soon!

**Comment by reiting** [comment]
@andrea-merrill @baltes The issue appears to be a github problem. Support is keeping an eye on it and going to try rerunning at intervals. If it isn't resolved by tomorrow, they will reach out to github support. This remains blocked, but like I said it IS done, just need to push the PR to staging.

**Comment by reiting** [comment]
@andrea-merrill @baltes okay Isaac has gone in and approved the PR for this. He said that since the unit test isn't an explicit failure it's fine. We are n ow waiting for the archive checks to run and once they do, I can merge! I'll keep you updated, but yay!

**Comment by reiting** [comment]
@baltes @andrea-merrill this finally got past all the checks and has been merged! There's not a super great way to review it, so I 'll add some screenshots below so you can see what it does:

VA File Num and no SSN:
![Image](https://github.com/user-attachments/assets/94acb679-88ef-49e7-a83e-7e3658553064)
![Image](https://github.com/user-attachments/assets/81c1593f-eb5a-4444-8674-e5eb2224c2a7)

SSN and no VA File Num:
![Image](https://github.com/user-attachments/assets/4f86596e-02fd-4c04-b408-3df5ca59abac)
![Image](https://github.com/user-attachments/assets/62516a68-4ebe-4d0a-a439-83c8f56371d2)

SSN AND VA File Num:
![Image](https://github.com/user-attachments/assets/891bed8a-6f9c-46fa-8149-46129137e40f)
![Image](https://github.com/user-attachments/assets/493e714c-73dc-4cbb-9250-f0c486c22c8c)

**Comment by baltes** [comment]
@reiting SSN is populating correctly in PEGA, but when only the file number is sent, then nothing is populating in the gui. That might just be that Cindy needs to update the mapping, but it looks like it also didn't make it onto the PDF. I'll do another test submit to double check.

<img width="1472" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOFlpQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--11ace230522e156f84610c21f445e31122843386/image.png" alt="image.png" />

**Comment by baltes** [comment]
Confirmed in a second submit. Let's make sure we're getting the file number on the PDF when only that is sent, otherwise we're losing data.

<img width="1472" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOHdpQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ff970436088d9cfc0384f294a493121ab609f023/image.png" alt="image.png" />

**Comment by reiting** [comment]
@baltes due to some stellar help from Michael, I've got a fix for this. It's currently in team review, soon to be in platform review.

**Comment by baltes** [comment]
@reiting Sorry if I missed an update in daily--does this still need a review on staging?

**Comment by reiting** [comment]
Not yet. It's still waiting on the 3 days for review from platform.

**Comment by reiting** [comment]
@baltes this is finally in staging and can be reviewed!!!!!

**Comment by baltes** [comment]
@reiting Needs another look I'm afraid. See examples below. This submit on staging includes SSN and Filenumber. Both should be mapped on the PDF, and file number should be populated in the UI. Everything else looks okay when entering one or the other, but we need to retain the file number in this scenario when both are entered.
![Image](https://github.com/user-attachments/assets/00764680-6423-48cb-b7dc-72b2711e13fa)
![Image](https://github.com/user-attachments/assets/72baa645-3319-4069-b4ab-fa083050e2d7)

**Comment by reiting** [comment]
@baltes okay, I can fix that, but the AC says "If SSN is populated AND file number is populated, send SSN in the metadata". But you want me to send BOTH in the metadata in that case, correct? I just want to be sure because I think I read it wrong.

**Comment by reiting** [comment]
@baltes I was talking with Michael about my PR, and he asked if we were supposed to send the va file number AS the ssn when it is the only thing available or if we were supposed to still keep sending it as va file number and just leave ssn empty. Can you clarify? I wasn't sure and I figured we'd just ask you so we can be positive!

**Comment by baltes** [comment]
@reiting No, if we have separate fields in the metadata we should keep the values distinct rather than overloading. As long as we're sending the available data we can work with Cindy as needed if its still not getting to the UI.

The other issue I was calling out is that in the case above the file number is lost because it's also not populated on the pdf. Maybe that was never working when both values are entered but in this case it will cause issues for processing.

**Comment by reiting** [comment]
@baltes okay this should be in staging and ready to review
